common/src/main/java/org/opensearch/ml/common/agent/MLToolSpec.java

```diff
@@ -24,20 +26,31 @@
 @EqualsAndHashCode
 @Getter
 public class MLToolSpec implements ToXContentObject {
+    public static final Version MINIMAL_SUPPORTED_VERSION_FOR_TOOL_CONFIG = CommonValue.VERSION_2_18_0;
+
     public static final String TOOL_TYPE_FIELD = "type";
     public static final String TOOL_NAME_FIELD = "name";
     public static final String DESCRIPTION_FIELD = "description";
     public static final String PARAMETERS_FIELD = "parameters";
     public static final String INCLUDE_OUTPUT_IN_AGENT_RESPONSE = "include_output_in_agent_response";
+    public static final String CONFIG_FIELD = "config";
 
     private String type;
     private String name;
     private String description;
     private Map<String, String> parameters;
     private boolean includeOutputInAgentResponse;
+    private Map<String, String> configMap;
 
     @Builder(toBuilder = true)
-    public MLToolSpec(String type, String name, String description, Map<String, String> parameters, boolean includeOutputInAgentResponse) {
+    public MLToolSpec(
+        String type,
+        String name,
+        String description,
+        Map<String, String> parameters,
+        boolean includeOutputInAgentResponse,
+        Map<String, String> configMap
```

Comment:
Is it valid that all the configs are Strings? Is it better to use Map<String, Object> here?

Reply:
+1 on this

Reply:
-1 on that.  The parameters map is already a `Map<String, String>`.  In the initial PR comment an example is given of a much more complex "object" but escaped as a String.

Trying to parse anything else from Flow Framework will get a lot more complex without typed objects to construct from the JSON.